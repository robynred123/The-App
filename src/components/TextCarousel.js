import React from "react";
import { ScrollView, View, Text, StyleSheet, Dimensions } from "react-native";

let screenWidth = Dimensions.get('window').width;

export const TextCarousel = (props) => {
  const { data } = props;

    return (
      <View>
        <ScrollView 
          horizontal 
          pagingEnabled={true}
          showsHorizontalScrollIndicator={true}
        >
          {data.map((item, i) => (
              <View key={item.key} style={styles.textContainer}> 
                <Text style={styles.textStyle}> 
                  {item.point} 
                </Text> 
              </View>
          ))}
        </ScrollView> 
      </View>
    );
}; 

const styles = StyleSheet.create({
  textContainer: {
    width: screenWidth,
    height: 150,
    justifyContent: 'center',
    alignItems: 'center',
    flexDirection: 'row'
  },
  textStyle:{
    textAlign: 'center',
    fontSize: 18
  }
})