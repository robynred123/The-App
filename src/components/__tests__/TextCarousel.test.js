import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'
import { TextCarousel } from '../TextCarousel';

Enzyme.configure({ adapter: new Adapter() })

const testData = [
    {
        'key': 1,
        'point': 'testing'
    },
    {
        'key': 2,
        'point': '12345'
    }
]

describe('<TextCarousel />', () => {

    it('should render the TextCarousel without crashing', () => {
        const component = shallow(<TextCarousel data={testData}/>)
        expect(component).toMatchSnapshot()
    })

    it('should render the TextCarousel with an array of data', () => {
        const component = shallow(<TextCarousel data={testData}/>)
        expect((component.find('Text')).map((node) => node.text())).toHaveLength(2)
        expect(component.find('Text').at(0).children().text()).toEqual('testing')
        expect(component.find('Text').at(1).children().text()).toEqual('12345')
        expect(component).toMatchSnapshot()
    })

})