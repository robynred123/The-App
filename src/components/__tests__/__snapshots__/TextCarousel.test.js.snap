// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextCarousel /> should render the TextCarousel with an array of data 1`] = `
<View>
  <ScrollViewMock
    horizontal={true}
    pagingEnabled={true}
    showsHorizontalScrollIndicator={true}
  >
    <View
      key="1"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 150,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        testing
      </Text>
    </View>
    <View
      key="2"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 150,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        12345
      </Text>
    </View>
  </ScrollViewMock>
</View>
`;

exports[`<TextCarousel /> should render the TextCarousel without crashing 1`] = `
<View>
  <ScrollViewMock
    horizontal={true}
    pagingEnabled={true}
    showsHorizontalScrollIndicator={true}
  >
    <View
      key="1"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 150,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        testing
      </Text>
    </View>
    <View
      key="2"
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": 150,
          "justifyContent": "center",
          "width": 750,
        }
      }
    >
      <Text
        style={
          Object {
            "fontSize": 18,
            "textAlign": "center",
          }
        }
      >
        12345
      </Text>
    </View>
  </ScrollViewMock>
</View>
`;
